agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -329,6 +334,34 @@ func (dg *dockerGoClient) time() ttime.Time {
 	return dg._time
 }
 
+// Pulls image manifest from the registry
+func (dg *dockerGoClient) PullImageManifest(
+	ctx context.Context, imageRef string, authData *apicontainer.RegistryAuthenticationData,
+) (registry.DistributionInspect, error) {
+	// Get auth creds
+	sdkAuthConfig, err := dg.getAuthdata(imageRef, authData)
+	if err != nil {
+		return registry.DistributionInspect{}, wrapPullErrorAsNamedError(imageRef, err)
```

Comment:
Q: Just curious, any reasons why we can't pass in `nil` if there were errors?

Reply:
Ah that's because this method has a value return type and not a pointer so the compiler won't allow returning `nil`. I just followed the method signature of Docker SDK which is 
```
DistributionInspect(ctx context.Context, imageRef, encodedRegistryAuth string) (registry.DistributionInspect, error)
```.

It is just like how functions that have a `(string, error)` return type would return `"", nil` for error cases and not a `nil`.